modules/ml/src/main/java/org/apache/ignite/ml/tree/DecisionTree.java

```diff
@@ -89,7 +92,7 @@
      * @param dataset Dataset.
      * @return Impurity measure calculator.
      */
-    abstract ImpurityMeasureCalculator<T> getImpurityMeasureCalculator(Dataset<EmptyContext, DecisionTreeData> dataset);
+    abstract ImpurityMeasureCalculator<T> getImpurityMeasureCalculator(Dataset<EmptyContext, DecisionTreeData> dataset, boolean useIndex);
```

Comment:
We already have `useIndex` as field, do we need to specify it as parameter as well?